connector/oidc/oidc.go

```diff
@@ -427,6 +446,26 @@ func (c *oidcConnector) createIdentity(ctx context.Context, identity connector.I
 		}
 	}
 
+	for _, cc := range c.newGroupsFromClaims {
+		newElement := ""
+		for _, clm := range cc.ClaimList {
+			// Non string claim value are ignored, concatenating them doesn't really make any sense
+			if claimValue, ok := claims[clm].(string); ok {
+				// Removing the delimiier string from the concatenated claim to ensure resulting claim structure
+				// is in full control of Dex operator
+				claimCleanedValue := strings.ReplaceAll(claimValue, cc.Delimiter, "")
```

Comment:
I wonder if this behavior should be controlled by another option (escapeDelimiter/cleanDelimiter)

Reply:
How about `escapeDelimiterCharacter` config key and if its not set the delimiter character will be removed?

Reply:
Hm, how would that work?

I mean, I doubt people want to _add_ anything like an escape character to their group names.

We can always add that functionality later should anyone need it. Personally, I don't see a use case for it at the moment.

Reply:
So just keep the current behaviour of removing any occurrences of the delimiter string ? or did misunderstood?

Reply:
`clearDelimiter: true` - remove delimiter as it is right now
`clearDelimiter: false (default)` - leave any delimiter untouched